src/utils/simulation.js

```diff
@@ -0,0 +1,210 @@
+import Person from "src/utils/person";
+import generateRandomnArray from "src/utils/generateRandomArray";
+import logger from "src/utils/logger";
+
+/** Class representing a simulation. */
+class Simulation {
+  /**
+   * @param {number} pop_size - The population size.
+   * @param {number} vacc_percentage - The percentage of population vaccinated.
+   * @param {number} initial_infected - The percentage of population vaccinated. 
+   * @param {object} virus - The virus object.
+   * @param {number} total_interactions - The total interactions every infected person will complete.
+   */
+  constructor(pop_size, vacc_percentage, initial_infected=0, virus, total_interactions) {
+    this.pop_size = pop_size;
+    this.vacc_percentage = vacc_percentage;
+    this.initial_infected = initial_infected;
+    this.virus = virus;
+    this.total_interactions = total_interactions;
+    this.logger = new logger();
+    this.newly_infected = [];
+    this.current_infected = [];
+    this.population = this.create_population();
+  }
+
+/**
+ * @param {number} id - The id of the current person infected.
+ * It filters through population for people who can still get sick.  
+ * @returns {array} - People who are not vaccinated and infected.
+ */
+  alivePeople(id) {
```

Comment:
since this is a getter function, maybe call it something like `getAlivePeople()`?

Reply:
Done